File: testcases/benchmark/student.cpp
struct Student {
    char name[50];
    float marks;
};
void process_students(Student* students, int count) {
    float sum = 0.0;
    for(int i = 0; i < count; i++) {
        sum += students[i].marks;
    }
    float avg = sum / count;
    for(int i = 0; i < count-1; i++) {
        for(int j = 0; j < count-i-1; j++) {
            if(students[j].marks < students[j+1].marks) {
                Student tmp = students[j];
                students[j] = students[j+1];
                students[j+1] = tmp;
            }
        }
    }
}

File: testcases/benchmark/student.rs
pub struct Student {
    pub name: String,
    pub marks: f32,
}
#[no_mangle]
pub fn process_students(students: &mut [Student]) {
    let sum: f32 = students.iter().map(|s| s.marks).sum();
    let avg = sum / students.len() as f32;
    students.sort_by(|a, b| b.marks.partial_cmp(&a.marks).unwrap());
    println!("Topper: {}", students[0].name);
}